solutions/standard/variables.tf

```diff
@@ -0,0 +1,222 @@
+  default     = 4096
+}
+
+variable "admin_pass" {
+  type        = string
+  description = "The password for the database administrator. If the admin password is null, the admin user ID cannot be accessed. More users can be specified in a user block."
+  default     = null
+  sensitive   = true
+}
+
+variable "users" {
+  type = list(object({
+    name     = string
+    password = string # pragma: allowlist secret
+    type     = string # "type" is required to generate the connection string for the outputs.
+    role     = optional(string)
+  }))
+  default     = []
+  sensitive   = true
+  description = "A list of users that you want to create on the database. Multiple blocks are allowed. The user password must be in the range of 10-32 characters. Be warned that in most case using IAM service credentials (via the var.service_credential_names) is sufficient to control access to the PostgreSQL instance. This blocks creates native postgres database users, more info on that can be found here https://cloud.ibm.com/docs/databases-for-postgresql?topic=databases-for-postgresql-user-management&interface=ui"
+}
+
+variable "service_credential_names" {
+  description = "The map of name and role for service credentials that you want to create for the database."
+  type        = map(string)
+  default     = {}
+}
+
+variable "resource_tags" {
+  type        = list(string)
+  description = "Optional list of tags to be added to the PostgreSQL instance."
+  default     = []
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the PostgreSQL instance created by the solution. [Learn more](https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial)."
+  default     = []
+}
+
+variable "configuration" {
+  description = "Database Configuration for PostgreSQL instance. Refer https://cloud.ibm.com/docs/databases-for-postgresql?topic=databases-for-postgresql-changing-configuration&interface=api for more details."
+  type = object({
+    shared_buffers             = optional(number)
+    max_connections            = optional(number)
+    max_prepared_transactions  = optional(number)
+    synchronous_commit         = optional(string)
+    effective_io_concurrency   = optional(number)
+    deadlock_timeout           = optional(number)
+    log_connections            = optional(string)
+    log_disconnections         = optional(string)
+    log_min_duration_statement = optional(number)
+    tcp_keepalives_idle        = optional(number)
+    tcp_keepalives_interval    = optional(number)
+    tcp_keepalives_count       = optional(number)
+    archive_timeout            = optional(number)
+    wal_level                  = optional(string)
+    max_replication_slots      = optional(number)
+    max_wal_senders            = optional(number)
+  })
+  default = null
```

Comment:
We need to find out if there are best practise settings here that we should set by default. Maybe we need to ask ICD and Dreadnought?

Reply:
I have asked this query in [ICD channel](https://ibm-cloudplatform.slack.com/archives/C534XRCF3/p1722006121403179), and in [Dreadnought-GoldenEye channel](https://ibm-cloudplatform.slack.com/archives/C042TNQU4MN/p1722006388006359)

Reply:
I am in conversation with @jojustin to get the default values.

Reply:
There were failures observed when setting the `wal_level` as `replica`. Checked with Joyce too and it appears the value `replica` will be available when once the replica server is set. Hence reverting this value back to `hot_standby` [Commented](https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/pull/450#issuecomment-2265337193) same here too.